lib/cloudregister/registerutils.py

```diff
@@ -313,18 +315,18 @@ def get_product_data(registration_target=None):
 
 
 # ----------------------------------------------------------------------------
-def credentials_files_are_equal(repo_credentials):
-    """Compare the base credentials files the the repo header and make
-       sure they have the same values."""
+def credentials_files_are_equal(repo_credentials, target_root='/'):
+    """Compare the base credentials file to the given file credentials file
+       for user name and password to check if they are equal."""
 
     if not repo_credentials or not isinstance(repo_credentials, str):
         return False
 
-    base_credentials_location = '/etc/zypp/credentials.d/'
-    target_root = get_zypper_target_root()
-    credentials_location = target_root + base_credentials_location
-    credentials_base = os.path.join(credentials_location, 'SCCcredentials')
-    credentials_header = os.path.join(credentials_location, repo_credentials)
+    credentials_location = os.path.join(target_root, ZYPP_CREDENTIALS_PATH)
+    credentials_base = os.path.join(credentials_location, BASE_CREDENTIALS_NAME)
+    credentials_header = os.path.join(
+        credentials_location, os.path.basename(repo_credentials)
```

Comment:
I'm not a big fan of os.path.join()  because it behaves weird in the following condition. Try this code

```python3
print(os.path.join('etc/zypp/credentials.d', '/foo'))

print(os.path.join('/etc/zypp/credentials.d', 'foo'))
```

Depending on the leading '/' the behavior is in my opinion not really a join. As you are using variables in your join parameters it is not guaranteed that the expected result comes out of this depending on if `credentials_location` and/or basename of repo_credentials comes up with a leading '/' or not. 

Because of this I avoid os.path.join and always use a combination of

```
os.path.normpath(os.sep.join([path_a, path_b]))
```

I leave it up to you but I think it's super risky to use os.path.join with dynamic variable content

Reply:
Thanks for pointing this out, I was unaware of the unexpected behavior of `os.path.join()`. AT present everything is consistent w.r.t. to the separators `/`. And this change makes the code more consistent of using `join()` vs. `"a" + "b"` which also depended on "a" and "b" having the separator properly included. I'll merge this and will come back with a follow up PR that replaces all uses of `join()` with the new pattern.